packages/shared/src/authz/fragments/workspaceSso.ts

```diff
@@ -21,64 +23,46 @@ export const maybeMemberRoleWithValidSsoSessionIfNeeded: AuthPolicyFragment<
   async ({ userId, workspaceId }) => {
     // Get workspace, so we can resolve its slug for error scenarios
     const workspace = await loaders.getWorkspace({ workspaceId })
-    if (workspace.isErr) {
-      switch (workspace.error.code) {
-        case 'WorkspaceNoAccess':
-        case 'WorkspaceNotFound':
-          return just(err(new WorkspaceNoAccessError()))
-        case 'WorkspaceSsoSessionNoAccess':
-          return just(err(workspace.error))
-        default:
-          throwUncoveredError(workspace.error)
-      }
-    }
+    // hides the fact, that the workspace does not exist
```

Comment:
considering that we're no longer running this in the FE and that loader results are cached I think we should make fragments as self-contained as possible to simplify using them - you should not need to prepare any checks before hand /have any assumptions about what the state of the workspace/policy is, before you invoke the fragment

so, judging by the figma and current usage of this, the fragment should 1) do the hasAnyWorkspaceRole check inside of it, instead of having to remember to invoke it before the check 2) maybe it should even accept a projectId as an arg, not a workspaceId, so you don't even have to do the `if (WORKSPACES_ENABLED && project.workspaceId)` check beforehand, and you can just invoke this fragment without wrapping it in any preparatory conditonals